build.zig

```diff
@@ -187,9 +177,23 @@ pub fn build(b: *std.build.Builder) void {
     }
 
     {
+        const simulator_options = b.addOptions();
+
+        const StateMachine = enum { testing, accounting };
+        simulator_options.addOption(
+            StateMachine,
+            "state_machine",
+            b.option(
+                StateMachine,
+                "simulator-state-machine",
+                "State machine.",
+            ) orelse .accounting,
+        );
+
         const simulator = b.addExecutable("simulator", "src/simulator.zig");
         simulator.setTarget(target);
         simulator.addOptions("vsr_options", options);
+        simulator.addOptions("simulator_options", simulator_options);
```

Comment:
Nit: could we call this `vsr_simulator_options`? Since it is a global identifier (kind of), it is nice to "namespace" within our library name.

Reply:
Yeah, a bit more consistent that way!

bors r+